lib/inputs/select-input.js

```diff
@@ -0,0 +1,92 @@
+'use strict';
+
+var _ = require('underscore');
+var BaseInput = require('./base-input');
+var errors = require('../errors');
+
+class SelectInput extends BaseInput {
+    constructor(id, params) {
+        super('select', id, params);
+    }
+
+    getInputDefaultValue() {
+        return this.params.items[0].value;
+    }
+
+    normalizeParams(params) {
+        params = Object.assign({}, params);
+
+        if (!params.items) {
+            throw errors.compileError('INPUT-REQ-PARAMS-MISSING', {
+                input_id: this.id,
+                missing_params: 'items'
+            });
+        }
+
+        if (!_.isArray(params.items)) {
+            throw errors.compileError('INPUT-INVALID-PARAM', {
+                input_id: this.id,
+                param: 'items',
+                param_value: params.items,
+                message: 'Item param must be an array'
+            });
+        }
+
+        params.items = params.items.map((item) => {
+            // if value is string, number or array expand to obj with value and label
+            if (_.isString(item) || _.isNumber(item) || _.isArray(item)) {
```

Comment:
Type checks for all Juttle values should use functions from `values.js` (`values.isString`, `values.isArray`, etc.). This is true here but in other places in the code too. There is no actual difference in behavior in the simple cases like numbers but it would still be good to do this as a matter of principle and to increase robustness should we later deal with Juttle-specific types such as moments.